js/views/migration_view.js

```diff
@@ -0,0 +1,179 @@
+;(function () {
+  'use strict';
+  window.Whisper = window.Whisper || {};
+
+  var State = {
+    DISCONNECTING: 1,
+    EXPORTING: 2,
+    COMPLETE: 3
+  };
+
+  Whisper.Migration = {
+    isComplete: function() {
+      return storage.get('migrationState') === State.COMPLETE;
+    },
+    inProgress: function() {
+      return storage.get('migrationState') > 0 || this.everComplete();
+    },
+    markComplete: function(target) {
+      storage.put('migrationState', State.COMPLETE);
+      storage.put('migrationEverCompleted', true);
+      if (target) {
```

Comment:
Realized that there are some places we call `markComplete()` just to affect the UI, and not because we actually just finished the export. In this case, we don't have the target location. So we do need this check.